### PR TITLE
Update foreign asset ids from v3 to v4

### PR DESCRIPTION
**Pull request by franciscoaguirre**
Addresses https://github.com/polkadot-fellows/runtimes/issues/209

Foreign assets in Asset Hub Kusama and Asset Hub Polkadot use v3 locations as ids. The newest version is v4 so this PR updates them.

No storage migration is needed since v3's and v4's encoding is the same.

Had to copy some test functions that took v3 locations into the repo until we update it to stable2409, when they actually switch to taking v4 locations.

### CODE REVIEW

File: rust-toolchain.toml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[toolchain]
+channel = "1.79.0"
```
Comment (acatangiu) [inline rust-toolchain.toml:2]: is this intended?

if so, could have been separate PR as it's unrelated

**Review by acatangiu — APPROVED** [review]

File: rust-toolchain.toml (hunk 1 above)
Comment (franciscoaguirre) [reply rust-toolchain.toml:2]: Ah I had to add it to get it to work locally. Will remove.

**Review by bkontur — APPROVED** [review]

File: integration-tests/emulated/tests/assets/asset-hub-kusama/src/tests/swap.rs [hunk 2]
```diff
@@ -21,14 +21,13 @@ use system_parachains_constants::kusama::currency::SYSTEM_PARA_EXISTENTIAL_DEPOS
 #[test]
 fn swap_locally_on_chain_using_local_assets() {
 	let asset_native = Box::new(
-		v3::Location::try_from(asset_hub_kusama_runtime::xcm_config::KsmLocation::get())
-			.expect("conversion works"),
+		asset_hub_kusama_runtime::xcm_config::KsmLocation::get().expect("conversion works"),
```
Comment (acatangiu) [inline integration-tests/emulated/tests/assets/asset-hub-kusama/src/tests/swap.rs:24]:
```suggestion
		asset_hub_kusama_runtime::xcm_config::KsmLocation::get(),
```

File: integration-tests/emulated/tests/assets/asset-hub-kusama/src/tests/swap.rs [hunk 3]
```diff
@@ -120,13 +119,12 @@ fn swap_locally_on_chain_using_local_assets() {
 #[test]
 fn swap_locally_on_chain_using_foreign_assets() {
 	let asset_native = Box::new(
-		v3::Location::try_from(asset_hub_kusama_runtime::xcm_config::KsmLocation::get())
-			.expect("conversion works"),
+		asset_hub_kusama_runtime::xcm_config::KsmLocation::get().expect("conversion works"),
```
Comment (acatangiu) [inline integration-tests/emulated/tests/assets/asset-hub-kusama/src/tests/swap.rs:122]:
```suggestion
		asset_hub_kusama_runtime::xcm_config::KsmLocation::get(),
```

File: integration-tests/emulated/tests/assets/asset-hub-kusama/src/tests/swap.rs [hunk 4]
```diff
@@ -120,13 +119,12 @@ fn swap_locally_on_chain_using_local_assets() {
...
 	);
 	let asset_location_on_penpal =
-		v3::Location::try_from(PenpalLocalTeleportableToAssetHub::get()).expect("conversion works");
+		PenpalLocalTeleportableToAssetHub::get().expect("conversion works");
```
Comment (acatangiu) [inline integration-tests/emulated/tests/assets/asset-hub-kusama/src/tests/swap.rs:125]:
```suggestion
		PenpalLocalTeleportableToAssetHub::get();
```

**Review by acatangiu — APPROVED** [review]

**Review by pandres95 — APPROVED** [review]
Quick question for runtime users/client developers: Is there a schedule for deprecating XCMv3 sometime soon?

**Comment by franciscoaguirre** [comment]
> Quick question for runtime users/client developers: Is there a schedule for deprecating XCMv3 sometime soon?

We're planning on removing v2 as soon as v5 comes out.
We could mark v3 as deprecated as soon as that happens.

**Pull request merged**